modules/lemmaDigitalBidAdapter.js

```diff
@@ -0,0 +1,573 @@
+        ua: navigator.userAgent,
+        language: (navigator.language || navigator.browserLanguage || navigator.userLanguage || navigator.systemLanguage),
+        w: (window.screen.width || window.innerWidth),
+        h: (window.screen.height || window.innerHeigh),
+        geo: {
+          country: params.country,
+          lat: params.latitude,
+          lon: params.longitude,
+          accuracy: params.accuracy,
+          region: params.region,
+          city: params.city,
+          zip: params.zip
+        },
+        ip: params.ip,
+        make: params.make,
+        model: params.model,
+        os: params.os,
+        carrier: params.carrier,
+        devicetype: params.device_type,
+        ifa: params.ifa,
+      };
+    }
+    return null;
+  },
+
+  /**
+   * create source object
+   */
+  _getSourceObject: (request) => {
+    var params = request && request.params ? request.params : null;
+    if (params) {
+      return {
+        pchain: params.pchain,
+        ext: {
+          schain: request.schain
+        },
+      };
+    }
+    return null;
+  },
+
+  /**
+  * get request ad sizes
+  **/
+  _getSizes: (request) => {
+    if (request.sizes && utils.isArray(request.sizes[0]) && request.sizes[0].length > 0) {
+      return request.sizes[0];
+    }
+    return null;
+  },
+
+  /**
+  * create the banner object
+  **/
+  _getBannerRequest: (bid) => {
+    var bObj;
+    var adFormat = [];
+    if (utils.deepAccess(bid, 'mediaTypes.banner')) {
+      var params = bid ? bid.params : null;
+      var bannerData = params && params.banner;
+      var sizes = spec._getSizes(bid) || [];
+      if (sizes && sizes.length == 0) {
+        sizes = bid.mediaTypes.banner.sizes[0];
+      }
+      if (sizes && sizes.length > 0) {
+        bObj = {};
+        bObj.w = sizes[0];
+        bObj.h = sizes[1];
+        bObj.pos = 0;
+        if (bannerData) {
+          bObj = utils.deepClone(bannerData);
+        }
+        sizes = bid.mediaTypes.banner.sizes;
+        if (sizes.length > 0) {
+          adFormat = [];
+          sizes.forEach(function (size) {
+            if (size.length > 1) {
+              adFormat.push({ w: size[0], h: size[1] });
+            }
+          });
+          if (adFormat.length > 0) {
+            bObj.format = adFormat;
+          }
+        }
+      } else {
+        utils.logWarn(LOG_WARN_PREFIX + 'Error: mediaTypes.banner.sizes missing for adunit: ' + bid.params.adunitId);
+      }
+    }
+    return bObj;
+  },
+
+  /**
+  * create the video object
+  **/
+  _getVideoRequest: (bid) => {
+    var vObj;
+    if (utils.deepAccess(bid, 'mediaTypes.video')) {
+      var params = bid ? bid.params : null;
+      var sizes = spec._getSizes(bid) || [];
+      if (sizes && sizes.length == 0) {
+        sizes = bid.mediaTypes && bid.mediaTypes.video ? bid.mediaTypes.video.playerSize : [];
+      }
+      if (sizes && sizes.length > 0) {
+        var videoData = params.video;
```

Comment:
Bidder params should the ad unit configurations. In this case, video params must first be read from AdUnit.mediaTypes.video when available.

Reply:
Made suggested changes.